opm/input/eclipse/Schedule/Schedule.cpp

```diff
@@ -1909,12 +1916,21 @@ namespace {
         }
 
         for (const auto& rst_group : rst_state.groups) {
-            const auto& group = this->snapshots.back().groups.get( rst_group.name );
+            auto& group = this->snapshots.back().groups.get( rst_group.name );
             if (group.isProductionGroup()) {
                 auto new_config = this->snapshots.back().guide_rate();
                 new_config.update_production_group(group);
                 this->snapshots.back().guide_rate.update(std::move(new_config));
-            }
+            } else if (group.isInjectionGroup()) {
```

Comment:
The simulator, strictly speaking, permits a group to have both production constraints (`GCONPROD`) and injection constraints (`GCONINJE`) with the `FIELD` group being the obvious example of a group which theoretically could be both.  If we want to cater to that possibility, then I think we should phrase this conditional as
```C++
if (group.IsProductionGroup()) {
    // ...
}
if (group.IsInjectionGroup()) {
    // ...
}
```
instead of using `else if` here.

Reply:
Good point, will update.